presto-main/src/test/java/com/facebook/presto/type/TestDecimalCasts.java

```diff
@@ -106,14 +106,33 @@ public void testDoubleToDecimalCasts()
         assertDecimalFunction("CAST(1000.0 AS DECIMAL(4,0))", decimal("1000"));
         assertDecimalFunction("CAST(1000.01 AS DECIMAL(7,2))", decimal("01000.01"));
         assertDecimalFunction("CAST(-234.0 AS DECIMAL(3,0))", decimal("-234"));
-        assertDecimalFunction("CAST(12345678901234567.0 AS DECIMAL(17,0))", decimal("12345678901234568"));
-        assertDecimalFunction("CAST(-12345678901234567.0 AS DECIMAL(17,0))", decimal("-12345678901234568"));
-        assertDecimalFunction("CAST(1234567890.0 AS DECIMAL(20,10))", decimal("1234567890.0000000000"));
-        assertDecimalFunction("CAST(-1234567890.0 AS DECIMAL(20,10))", decimal("-1234567890.0000000000"));
+        assertDecimalFunction("CAST(1234567890123456.0 AS DECIMAL(16,0))", decimal("1234567890123456"));
```

Comment:
can you add tests like
`assertDecimalFunction("CAST(1234567890123456.9 AS DECIMAL(16,0))", decimal("1234567890123457"));`
`assertDecimalFunction("CAST(-1234567890123456.9 AS DECIMAL(16,0))", decimal("-1234567890123457"));`
to test rounding?